index.js

```diff
@@ -267,10 +281,16 @@ ShareDbMongo.prototype._writeSnapshot = function(collectionName, id, snapshot, o
         callback(null, true);
       });
     } else {
-      collection.replaceOne({_id: id, _v: doc._v - 1}, doc, function(err, result) {
-        if (err) return callback(err);
-        var succeeded = !!result.modifiedCount;
-        callback(null, succeeded);
+      request.query = {_id: id, _v: request.doc._v - 1};
+      self.trigger(self.MIDDLEWARE_ACTIONS.beforeEdit, request, function(middlewareErr) {
```

Comment:
No it does not at the moment, and I believe that if we wanted to, we should make another middleware for `beforeCreate`. They are different because `beforeEdit` has a query (for an existing document) but `beforeCreate` would not.

Now that you mention it though, we probably do want to add the `beforeCreate` for our purposes to ensure we never create documents with missing shard keys 🤔

Reply:
But I would suggest let's leave the `beforeCreate` for now - I will add it in another PR if we need it